go/chat/signencrypt/codec.go

```diff
@@ -450,17 +462,24 @@ func (r *codecReadWrapper) Read(callerBuf []byte) (int, error) {
 	return 0, io.EOF
 }
 
-func NewEncodingReader(encKey SecretboxKey, signKey SignKey, nonce Nonce, innerReader io.Reader) io.Reader {
+// NewEncodingReader creates a new streaming encoder.
+// The signaturePrefix argument is prepended before signing anything.
```

Comment:
This line kind of sounds like we're prepending the prefix directly to the message, which isn't what we're doing. I'd call what we're doing "signing the prefix alongside the message hash", but maybe we don't need to describe it at all here and just focus on the requirements. Totally your call.